frontend/src/concepts/pipelines/topology/usePipelineTaskTopology.ts

```diff
@@ -0,0 +1,96 @@
+import * as React from 'react';
+import { createNode } from '~/concepts/pipelines/topology/core/utils';
+import {
+  PipelineRunKind,
+  PipelineRunTask,
+  PipelineRunTaskParam,
+  PipelineRunTaskWhen,
+} from '~/k8sTypes';
+
+const getNameFromTaskRef = (taskRef: string) => {
+  const match = taskRef.match(/\$\(tasks\.([a-z0-9-]+)\./);
```

Comment:
😠 Kubeflow makes use of Tekton's "we'll run in the order of your properties" and doesn't definite the directed graph... so I have to reverse engineer it from the references to tasks -- which is ick... but we don't have a choice.